src/components/Layout/footer/footer.js

```diff
@@ -1,7 +1,35 @@
+'use client'
 import Link from 'next/link'
 import { navigation } from './config'
+import { useState } from 'react'
 
 export function Footer() {
+  const [email, setEmail] = useState('')
+  const handleSubmitEmail = async (e) => {
+    e.preventDefault()
+    try {
+      const response = await fetch('/api/subscribe', {
+        method: 'POST',
+        body: JSON.stringify({
+          email: email,
+        }),
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      })
+      console.log(response)
+      if (response.ok) {
+        console.log('Subscription successful!')
+        setEmail('')
+      } else {
+        console.log('Subscription failed. Please try again.')
+      }
+    } catch (error) {
+      console.log(error)
+      console.log('An error occurred. Please try again.')
```

Comment:
Should I add toastify as it is in Legaliser?

Reply:
yes please